iaso/api/payments/views.py

```diff
@@ -0,0 +1,152 @@
+    filter_backends = [
+        filters.OrderingFilter,
+        django_filters.rest_framework.DjangoFilterBackend,
+        potential_payment_filters.UsersFilterBackend,
+        potential_payment_filters.UserRolesFilterBackend,
+        potential_payment_filters.FormsFilterBackend,
+        potential_payment_filters.ParentFilterBackend,
+        potential_payment_filters.StartEndDateFilterBackend,
+    ]
+    ordering_fields = [
+        "user__username",
+        "user__last_name",
+        "user__first_name",
+        "created_at",
+        "updated_at",
+        "status",
+        "created_by__username",
+        "updated_by__username",
+        "change_requests",
+    ]
+    serializer_class = PotentialPaymentSerializer
+
+    results_key = "results"
+    http_method_names = ["get", "head", "options", "trace"]
+
+    def get_queryset(self):
+        return (
+            PotentialPayment.objects.prefetch_related("change_requests")
+            .filter(change_requests__created_by__iaso_profile__account=self.request.user.iaso_profile.account)
+            .distinct()
+        )
+
+    @swagger_auto_schema(auto_schema=None)
+    def retrieve(self, request, *args, **kwargs):
+        raise NotFound("Retrieve operation is not allowed.")
+
+    @swagger_auto_schema(
+        manual_parameters=[
+            openapi.Parameter(
+                name="users",
+                in_=openapi.IN_QUERY,
+                description="A comma-separated list of User IDs associated with the payments",
+                type=openapi.TYPE_STRING,
+            ),
+            openapi.Parameter(
+                name="user_roles",
+                in_=openapi.IN_QUERY,
+                description="A comma-separated list of User Role IDs associated with the payments",
+                type=openapi.TYPE_STRING,
+            ),
+            openapi.Parameter(
+                name="parent_id",
+                in_=openapi.IN_QUERY,
+                description="The ID of the parent organization unit linked to the change requests. This should also include child units.",
+                type=openapi.TYPE_INTEGER,
+            ),
+            openapi.Parameter(
+                name="change_requests__created_at_after",
+                in_=openapi.IN_QUERY,
+                description="The start date for when the change request has been validated. Format: YYYY-MM-DD",
+                type=openapi.TYPE_STRING,
+                format=openapi.FORMAT_DATE,
+            ),
+            openapi.Parameter(
+                name="change_requests__created_at_before",
+                in_=openapi.IN_QUERY,
+                description="The end date for when the change request has been validated. Format: YYYY-MM-DD",
+                type=openapi.TYPE_STRING,
+                format=openapi.FORMAT_DATE,
+            ),
+        ]
+    )
+    def list(self, request):
+        orders = request.GET.get("order", "user__last_name").split(",")
```

Comment:
I don't think this is gonna be taken into consideration, if you look at the code for `filters.OrderingFilter`.

```
def get_ordering(self, request, queryset, view):
        """
        Ordering is set by a comma delimited ?ordering=... query parameter.

        The `ordering` query parameter can be overridden by setting
        the `ordering_param` value on the OrderingFilter or by
        specifying an `ORDERING_PARAM` value in the API settings.
        """
        params = request.query_params.get(self.ordering_param)
        if params:
            fields = [param.strip() for param in params.split(',')]
            ordering = self.remove_invalid_fields(queryset, fields, view, request)
            if ordering:
                return ordering
```

So I guess if you want to override it you should maybe do something add something like this on line 76

`ordering = ['user__last_name']`

and then remove line 129 

(To be tested)